common/web/types/test/ldml-keyboard/test-unicodeset-parser-api.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ * 
+ * Created by Dr Mark C. Sinclair on 2024-11-28
+ * 
+ * Test code for unicodeset-parser-api.ts
+ */
+
+import 'mocha';
+import { assert } from 'chai';
+import { UnicodeSet } from '../../src/ldml-keyboard/unicodeset-parser-api.js';
+
+class MockUnicodeSet extends UnicodeSet {
+  constructor(public pattern: string, public ranges: number[][]) {
+    super(pattern, ranges); // does nothing
+    this.pattern = pattern;
+    this.ranges  = ranges;
```

Comment:
This shouldn't be necessary as you have `public` in the constructor properties, which does the same thing.

```suggestion
```

Reply:
Now with improved knowledge of TS, I can take out the whole `MockUnicodeSet`.